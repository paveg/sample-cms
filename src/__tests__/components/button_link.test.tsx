import * as React from 'react';
import renderer from 'react-test-renderer';
import ButtonLink from '../../components/button_link';

const testHref = '/test';
const testText = 'TEST';

describe('ButtonLink', () => {
  describe('when normal call', () => {
    const component = renderer.create(<ButtonLink href={testHref}>{testText}</ButtonLink>);
    it('renders correctly component', () => {
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('with small props', () => {
    const component = renderer.create(
      <ButtonLink href={testHref} small>
        {testText}
      </ButtonLink>,
    );
    it('renders correctly component', () => {
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
