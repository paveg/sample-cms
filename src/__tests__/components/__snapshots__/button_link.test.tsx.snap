// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink when normal call renders correctly component 1`] = `
<a
  aria-disabled={false}
  className="MuiButtonBase-root MuiButton-root MuiButton-text"
  href="/test"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
>
  <span
    className="MuiButton-label"
  >
    TEST
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</a>
`;

exports[`ButtonLink with small props renders correctly component 1`] = `
<a
  aria-disabled={false}
  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
  href="/test"
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
>
  <span
    className="MuiButton-label"
  >
    TEST
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</a>
`;
